### PR TITLE
Allow [tt] for pass

### LINKED ISSUE #587: sgf parser explosion
i don't have a callstack handy as i was using Seth's box.
But i was using [this sgf](https://pandanet-igs.com/system/sgfs/8031/original/AQEYAWDNSF.sgf?1543413330)

from [this page](https://pandanet-igs.com/communities/agacityleague/rounds/518)

### PR DESCRIPTION
**Pull request by sethtroisi**
Fixes #587

### CODE REVIEW

File: cc/coord.cc [hunk 1]
```diff
@@ -30,7 +30,7 @@ namespace {
 
 Coord TryParseKgs(absl::string_view str) {
   auto upper = absl::AsciiStrToUpper(str);
-  if (upper == "PASS") {
+  if (upper == "PASS" || upper == "TT") {
```
Comment (tommadams) [inline cc/coord.cc]: I don't think this is correct: `TT` isn't a valid KGS coordinate. The GTP and KGS docs I've found all say that pass should be represented as the string `pass`. I think you can revert this particular line.

File: cc/coord.cc [hunk 2]
```diff
@@ -52,7 +52,7 @@ Coord TryParseKgs(absl::string_view str) {
 }
 
 Coord TryParseSgf(absl::string_view str) {
-  if (str.empty()) {
+  if (str.empty() || str == "tt") {
```
Comment (amj) [inline cc/coord.cc]: Cast to lowercase first?

Comment (tommadams) [reply cc/coord.cc]: Uppercase letters are reserved for representing points 27-52 on larger boards, see: https://www.red-bean.com/sgf/go.html#types

Comment (tommadams) [reply cc/coord.cc]: To be clear, assuming lowercase is the correct thing to do here.

File: cc/coord.cc (hunk 1 above)
Comment (amj) [reply cc/coord.cc]: i think "KGS" is the wrong name for these coordinates.  these are GTP coordinates and we're using this to translate our GTP commands.

So i think this should be "TryParseGtp"

File: cc/coord.cc (hunk 2 above)
Comment (amj) [reply cc/coord.cc]: i guess the right thing to do is to compare `(str == "tt" && boardsize ==19)`

**Pull request merged**